### ISSUE TITLE
Support for weatherapi.com

### ISSUE THREAD

**Issue by elerch** [chubin/wttr.in#506]
I noticed that from https://www.worldweatheronline.com/developer/api/ there is mention of a free API. This link takes you to another site, https://www.weatherapi.com/ which appears to be the same company. This second API provides a perpetual free tier for 20k requests per month which should be reasonable for a personal use self hosted wttr.in.

The API, however, seems to be a vastly different data format, with two different calls needed for current weather/forecast data. It also appears to be missing hourly data. If you're willing to forego hourly data it's possible to have a completely free self-hosted solution with https://www.weatherapi.com/, but it would require some changes in this project to manage the alternate data shape.

**Comment by chubin** [comment]
Yes, agree. There are several popular different sources (sometimes even free), that we could use.
We could gather a list of such services, and think about some modular structure, or better to say
data format for their support.

For the beginning we could transform all the data from different sources into the format that wttr.in
already uses, maybe though this format is not ideal, and should be changed into something new. But the core idea stays the same: for each new data source, a transformation file will be needed, that will become a hierarchical data structure and return a hierarchical data structure as output.

Maybe we need some generalized approach here: some data transformation tables, like jq, but for the whole data.
I think we should just start and see.

Your recent PR was a very big and useful step in this direction
